### ISSUE TITLE
Broken implementation

### ISSUE THREAD

**Issue by jorgerobles** [LaserWeb/ReglTests#1]

**Comment by jorgerobles** [comment]
@tbfleming Despite Cam issues (if there's something on console please attach, some people tried it and seems to work for them too) 

There is tests with the @DarklyLabs emblaser perspective instead of video (the video part is commented for labbing)

I've got a huge (total) lack of knowledge about WebGL.

I've ported some commands from several sources to Regl (Barrel from https://jywarren.github.io/fisheyegl/example/ and Perspective from glfx/perspective )
They are working well individually, as well as demo are working. But I cannot combine them into the  pipeline.

* The main issue is the "textures" are flipped horizontally so any work on perspective transform seems mangled.

* The second issue is that passing the barrel over perspective is not working well. I think the input buffer should be clipped so perspective can work well without barrel corner spikes

* The third is Mipmapping (output is awful)

**Comment by tbfleming** [comment]
I think barrel should be done first, then inverse perspective. They're inverse transforms, so should be applied in reverse order of what happens to the real-world image. First, the image arrives at the camera in perspective, then the camera's lens distorts it.

**Comment by jorgerobles** [comment]
Could you take a look to https://github.com/LaserWeb/ReglTests/blob/master/src/draw-commands/commands.js `drawCommand`? I don't understand why it gets flipped horizontally.

**Comment by tbfleming** [comment]
```
gl_Position = vec4(1.0 - 2.0 * position, 0, 1);
```

This flips both X and Y and converts from the range [0,1] to [-1,1]. Y needs to be flipped because WebGL expects images to be bottom-up. Try:

```
gl_Position = vec4(-1.0 + 2.0 * position.x, 1.0 - 2.0 * position.y, 0, 1);
```

**Comment by jorgerobles** [comment]
Thaaanks :)

**Comment by jorgerobles** [comment]
I've uploaded the best I could achieve. But is crappy. It looks awful :(

**Comment by openhardwarecoza** [comment]
Can we close this issue?

**Comment by jorgerobles** [comment]
Yes, indeed the whole branch could be deleted. And we have plenty of webgl
issues elsewhere :)

**Comment by openhardwarecoza** [comment]
Cool.  (: Delete it when you get a chance (: (Maybe even the whole repo?)
